peakina/readers/excel.py

```diff
@@ -0,0 +1,199 @@
+    NEW = "new"
+    OLD = "old"
+
+
+def _old_xls_rows_iterator(
+    wb: Any, sh_name: str, preview: Optional[PreviewArgs]
+) -> Generator[Any, Any, Any]:
+
+    if preview:
+        to_iter = range(preview.offset, preview.offset + preview.nrows)
+    else:
+        to_iter = range(wb[sh_name].nrows)
+
+    for rx in to_iter:
+        yield wb[sh_name].row(rx)
+
+
+def _new_xls_rows_iterator(
+    wb: Any, sh_name: str, preview: Optional[PreviewArgs]
+) -> Generator[Any, Any, Any]:
+
+    yield wb[sh_name].iter_rows(
+        min_row=preview.offset if preview else None,
+        max_row=preview.offset + preview.nrows if preview else None,
+        values_only=True,
+    )
+
+
+def _get_rows_iterator(
+    wb: Any, excel_type: EXCEL_TYPE, sheet_name: str, preview: Optional[PreviewArgs]
+) -> Generator[Any, Any, Any]:
+
+    if excel_type is EXCEL_TYPE.OLD:
+        return _old_xls_rows_iterator(wb, sheet_name, preview)
+
+    return _new_xls_rows_iterator(wb, sheet_name, preview)
+
+
+def _build_row_subset(
+    row: Union[List[Any], Tuple[Any]],
+    sh_name: str,
+    sheetnames: List[str],
+    row_number: int,
+    row_subset: List[str],
+) -> Tuple[int, List[str]]:
+
+    cells = [str(cell.value) if type(cell) not in [str, int, float] else str(cell) for cell in row]
+
+    if len(sheetnames) > 1:
+        # TO add the column names at the top
+        if row_number == 0:
+            row_subset.append(f'{",".join([*cells, "__sheet__"])}\n')
+        else:
+            row_subset.append(f'{",".join([*cells, sh_name])}\n')
+    else:
+        row_subset.append(f'{",".join(cells)}\n')
+
+    row_number += 1
+
+    return row_number, row_subset
+
+
+def _get_row_subset_per_sheet(
+    wb: Any,
+    sh_name: str,
+    sheetnames: List[str],
+    preview: Optional[PreviewArgs],
+    excel_type: EXCEL_TYPE,
+    row_subset: List[str],
+    skiprows: Optional[int] = None,
+    nrows: Optional[int] = None,
+) -> List[str]:
+
+    row_iterator = _get_rows_iterator(wb, excel_type, sh_name, preview)
+
+    if excel_type is EXCEL_TYPE.NEW:
+        for row_number, gen in enumerate(row_iterator):
+            for row in gen:
+                if skiprows:
+                    if row_number < skiprows:
+                        continue
+                row_number, row_subset = _build_row_subset(
+                    row, sh_name, sheetnames, row_number, row_subset
+                )
+                if nrows:
+                    if row_number == nrows:
+                        break
+    else:
+        for row_number, row in enumerate(row_iterator):
+            if skiprows:
+                if row_number < skiprows:
+                    continue
+            row_number, row_subset = _build_row_subset(
+                row, sh_name, sheetnames, row_number, row_subset
+            )
+            if nrows:
+                if row_number == nrows:
+                    break
+    return row_subset
+
+
+def _read_sheets(
+    wb: Any,
```

Comment:
Instead of `Any`: `Union[openpyxl.workbook.Workbook, xlrd.book.Book]`

Reply:
Thanks, 👍🏾 Updated !